### DISCUSSION TITLE
Exteral clock instead of avr timer?

### DISCUSSION THREAD

**Discussion by ziotibia81** [udoklein/dcf77#44] [Ideas]
Hello and happy new year! 

My scenario is: DIY clock, rtc driven and DCF77 for initial sync and periodic time check. Environment is electrically noisy so standard dcf77 library doesn't sync, but dcf77_xtal does.

Rtc is a DS3231 so I have a SQW pin able to output 1,024khz derived from TCXO.

Injecting this signal to an arduino input and use interrupt from this instead of timer interrupt could be usable? It my work on board with cheap resonator...

**Comment by udoklein** [comment]
Yes, this is doable. You will need to implement a suitable 1 kHz clock generator. Obviously this is not trivial with a 1024 Hz clock. However the DS3231 can also generate a 32768 clock. Dividing this down to 1 kHz is significantly easier. Dividing down to 1 kHz will of course introduce some phase jitter but this will be below 3%. This will not degrade the performance of the lock.

**Comment by ziotibia81** [comment]
Do you mean divide 32768 by 33 (992,97 Hz; 1s every 993 tick) or something more complex to get a more precise freqency?

**Comment by udoklein** [comment]
I am proposing something along the following line:
[32768_Hz_experiment](https://gist.github.com/udoklein/f95a6894b2a43c9c4878b57684a93856.js)](https://gist.github.com/udoklein/f95a6894b2a43c9c4878b57684a93856)

However I have to say this was created as an experiment and never fully tested. Hence I did not release it to the general public. If it works for you: fine, please provide feedback. If it does not work: please fix it and tell me what you did.
